File: Neetcode-150-js/Arrays and Hashing/top-k-frequent-elements.js

// https://leetcode.com/problems/top-k-frequent-elements/

/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function(nums, k) {
  const obj = {};
  for(const num of nums){
    if(obj[num]){
      obj[num].count++;
    }else obj[num] = {'num' : num, 'count' : 1};
  }
  
  // console.log(obj);
  const topK = Array.from(Object.values(obj))
  .sort((a, b) => b.count - a.count)
  .splice(0, k)
  .map(ele => ele.num);
  
  // console.log(topK);
  return topK;
};

File: Striver Important Interview Questions/Day 12 - Heaps/top-k-frequent-elements.cpp

// https://leetcode.com/problems/top-k-frequent-elements/description/

class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<int> ans;

        unordered_map<int, int> mp;
        for(auto ele : nums) mp[ele]++;

        priority_queue<pair<int, int>> pq;
        for(auto it : mp){
            pq.push({it.second, it.first});
        }

        while(k--){
            auto top = pq.top();
            pq.pop();
            ans.push_back(top.second);
        }

        return ans;
    }
};